### DISCUSSION TITLE
New complication

### DISCUSSION THREAD

**Discussion by kcossabo** [Blackymas/NSPanel_HA_Blueprint#778] [General]
Well, I upgraded, and bad things...

Home Assistant 2023.5.0
Supervisor 2023.04.1
Operating System 10.1
Frontend 20230503.1 - latest
Blueprint Version: v.3.2.2
Firmware: 2023.4.4 (May 3 2023, 19:52:59)
[logs_nspanelone_logs (1).txt](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/11392216/logs_nspanelone_logs.1.txt)

The panel is on the HA blue screen, with IP address correct,TFT 3.2.2, ESPHome 3.3, Blueprint (Blank)
Does not leave that screen. So far I have
- powered on and off
- ESPHome was showing an upgrade, so I upgraded
- Integrations -> ESPHome -> Update TFT display (It updated)

How do I troubleshoot?

**Comment by Herbert69-Cyber** [comment]
Same problem here, bit like the Microsoft famous blue screen of death. I think it was the ESphome update 2 days ago that broke it. Waiting patiently for a fix.

**Reply by Herbert69-Cyber** [reply]
Sorry just found fix documented in issues log

**Comment by edwardtfn** [comment]
> The panel is on the HA blue screen, with IP address correct,TFT 3.2.2, ESPHome 3.3, Blueprint (Blank)

The problem here is that with the update of your ESPHome, it recompile each of your devices and for your panel that means getting the latest code from GitHub, which is now v3.3, but it wasn't the same for the other parts (TFT file and the Blueprint), so you have a version mismatch which will cause your panel to be stuck in the boot page (the blue screen) or on "Loading...".

You can fix that by updating the TFT file and the Blueprint also to v3.3, so they will have the same version as your ESPHome.

There are some update procedures on wiki, but please let me know if you get stuck on that process and I will try to help:
https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-HowTo---All-important-thing-you-should-know#2-update-blueprint

**Reply by kcossabo** [reply]
Thank You - still working the issue but closer. I added the above to the tips needed.

Current situation;

Downloaded new blueprint
Rebooted HA

<Panel now has black with swirl screen showing 'loading'>

- change date on config -> no change
- Power on and off -> no change

Updated TFT Display

- Panel shows downloading
- completes, does check 
- completes
- reboots

<same condition black with swirl screen showing 'loading'>

<tested physical buttons, operating according to the blueprint>

Sent a Reset from _**Settings --> Devices & Services --> Integrations**_

<same condition black with swirl screen showing 'loading' physical buttons, operating according to the blueprint>

- TFT: 3.2.2
- ESPHome 3.3
- BluePrint 3.3

Went to ESP Home tab, and update all
Then the panel, install wirelessly

<same condition black with swirl screen showing 'loading' physical buttons, operating according to the blueprint>

Sent a Reset from _**Settings --> Devices & Services --> Integrations**_

<same condition black with swirl screen showing 'loading' physical buttons, operating according to the blueprint>

- change date on config -> no change
- Power on and off -> 
[logs_update-all.txt](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/11397707/logs_update-all.txt)

[logs_nspanelone_run.txt](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/11397704/logs_nspanelone_run.txt)
no change
![Subject](https://user-images.githubusercontent.com/6267080/236211809-58f6725d-4c2e-45d8-9724-7eaf5358b667.png)
![IMG_6582 copy](https://user-images.githubusercontent.com/6267080/236212199-7dbf0a05-22a8-47bf-87e7-864904b945b8.png)

**Reply by kcossabo** [reply]
ESPHome Log
[5c53de3b_esphome_2023-05-04T13-02-58.251Z.log](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/11397737/5c53de3b_esphome_2023-05-04T13-02-58.251Z.log)

**Reply by edwardtfn** [reply]
You have to update TFT to v3.3.

Download the file from GitHub:
- [EU version](https://github.com/Blackymas/NSPanel_HA_Blueprint/raw/main/nspanel_eu.tft)
- [US version](https://github.com/Blackymas/NSPanel_HA_Blueprint/raw/main/nspanel_us.tft)

Then upload it to your local www server in Home Assistant using the File Editor (browse to the folder www, delete the old TFT from there and then upload the new one).
Then go to your device page (Settings > Devices and Services > ESPHome) and press `Upload TFT`.

https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-HowTo---All-important-thing-you-should-know#4-update-tft

**Reply by kcossabo** [reply]
Cool, did not read to download the new version, and the versions looked like they were 2 weeks old. so I left it like it is.

I cloned (this morning) the git repository to the Pi, but the link above 

> https://github.com/Blackymas/NSPanel_HA_Blueprint/raw/main/nspanel_us.tft

is not in my clone.... no /raw/...


anyways, did the download, now up and running. On boot screen;


- TFT: 3.3
- ESPHome 3.3
- BluePrint 3.3

Also two new icons in the upper left, a 1 and a 2 in a box?



![Subject 2](https://user-images.githubusercontent.com/6267080/236217384-d69a6e25-6ec6-438f-9e90-2cd607f92e34.png)

# Thank You

--- 

Do I have this correct?

Process;
- download and update Blueprint
- download and update TFT file (not sure where this is for each release)
- go to the ESPHome Tab, Update all, then update the panel

**Reply by edwardtfn** [reply]
> Also two new icons in the upper left, a 1 and a 2 in a box?

That shows physical relays 1 and 2 are `on`.

**Comment by Herbert69-Cyber** [comment]
Thank you,

I have updated TFT: 3.3 and ESPHome: 3.3 and the Blueprint to 3.3 but it is
still stuck on the boot screen

As far as I can see all three are the same version, even the relay switches
no longer work even on fullback.  I'm using the US Version by the way.

Clinton

**Reply by edwardtfn** [reply]
> I have updated TFT: 3.3 and ESPHome: 3.3 and the Blueprint to 3.3 but it is
still stuck on the boot screen

Can you see in the boot page the version of all 3 parts (TFT, ESPHome and Blueprint)? I wanna know if all parts have reported it's new versions to the panel.
By the way, what if you press the reboot button on that screen?
Any error message on your Home Assistant logs? (Settings > System > Logs)

**Comment by Herbert69-Cyber** [comment]
Blueprint is blank, TFT and ESphome is populated with 3.3.

Get this event error, but that is all

Thanks agian

[image: image.png]

**Reply by edwardtfn** [reply]
I didn't got the image. Could you please share that again?

Also, could you pleas eco firm you can see version 3.3 whe editing settings in your blueprint? And as you get to that screen, please double check if the panel is selected in the first field.
![IMG_6206](https://user-images.githubusercontent.com/94725493/236166972-cb776c14-4e91-4e68-950c-316261580b3e.jpeg)

**Comment by Herbert69-Cyber** [comment]
Yes when I edit the blue print I can see version 3.3.

Hopefully screen shot works this time.

I suspect my problem is the blueprint, I have 2 panels each attached to a
different ESp32 device in HA and I also have 2 automations in HA perhaps
some conflict going on. Both devices are stuck on the boot screen

Solution might be to delete automations and blueprint and start again.

Clinton

**Reply by edwardtfn** [reply]
> Hopefully screen shot works this time.

Unfortunately still no image. I belive it won't work when you send via email.

> I suspect my problem is the blueprint, I have 2 panels each attached to a
different ESp32 device in HA and I also have 2 automations in HA perhaps
some conflict going on. Both devices are stuck on the boot screen

It could be the Blueprint. It's hard to say with the info I have so far.
Having multiple panels shouldn't be a problem. I have 3 panels at home, both US and EU models, and they are all working fine.

> Solution might be to delete automations and blueprint and start again.


That's always n efficient way to fix issues, but a bit radical. But if that is easy for you to try, why not? 🙂

**Comment by Herbert69-Cyber** [comment]
Here is the error message plain txt from an hour ago when I tried to reboot

 to {"panspanel NSPanel event changed to {"page": "boot", "component":
"currentpage", "value": "pagechange"}

Off to bed now its 10:20pm in New Zealand, tackle again tomorrow

Clinton
ge": "boot", "component": "currentpage", "value": "pagechange"}

**Reply by edwardtfn** [reply]
> to {"panspanel NSPanel event changed to {"page": "boot", "component": "currentpage", "value": "pagechange"}

> Clinton ge": "boot", "component": "currentpage", "value": "pagechange"}

You message is a bit truncated. Could you please use [this GitHub link](https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/778) to upload the screen shots?


> Off to bed now its 10:20pm in New Zealand, tackle again tomorrow

Sure!
Rest well. 🙂

**Reply by Herbert69-Cyber** [reply]
I only have this message in the Logbook and when I look at the ESPHome log all the sensors are reporting. So I think my issue is definitely to do with the blueprint maybe wrong path

clinton

![image](https://user-images.githubusercontent.com/91824604/236325966-fa421664-f362-452d-a0ec-9a22820600d3.png)

**Reply by edwardtfn** [reply]
Well, based on your screenshot, the problem is that your ESPHome is not connected to Home Assistant... Maybe a temporary network issue, but I would try to remove the panel integration and add it again.

But when I was asking for a log, is was not the logbook, it was the log you can see under Settings > System > Logs.

**Reply by Herbert69-Cyber** [reply]
Maybe this is it, found these entries in the log

![image](https://user-images.githubusercontent.com/91824604/236381231-8c66e94f-daea-4450-87ef-46006468a79c.png)

**Reply by edwardtfn** [reply]
Well, the first Al last messages are related to some communication problem with your panel, but I think the main issue is in the message in the middle, saying you have 2 automations with the same id. You will have to remove one of those or at least change its ID (but then make sure you don't have 2 automations assigned to the same panel).

**Reply by Herbert69-Cyber** [reply]
Had to delete my automatons and the blueprint and reload but got there thanks to your hint.  On the bright side I have the weekend to reconfigure my switches :-)  
![IMG20230505214006](https://user-images.githubusercontent.com/91824604/236426031-9d7f567c-87c9-4a82-ab22-10936ff9e8f2.jpg)

**Reply by Herbert69-Cyber** [reply]
P.s its not actually raining at the moment lol

**Reply by edwardtfn** [reply]
This info comes from your weather integration.
